collateral.md

```diff
@@ -26,6 +32,63 @@ module COLLATERAL
         </flips>
       </collateral>
 
+    syntax MCDContract ::= GemContract
+    syntax GemContract ::= "Gem" Address
+    syntax MCDStep ::= GemContract "." GemStep [klabel(gemStep)]
+ // ------------------------------------------------------------
+    rule contract(Gem ACCTGEM . _) => Gem ACCTGEM
+    rule [[ address(Gem ACCTGEM) => ACCTGEM ]] <gem-addr> ACCTGEM </gem-addr>
```

Comment:
Is there any reason to have `<gem-addr> ACCTGEM </gem-addr>` in here? Or can we just do:

```
rule address(Gem ACCTGEM) => ACCTGEM
```

Actually, this seems kinda strange to me anyway. In `Flip_`, the extra parameter is the `Ilk` identifier, _not_ the contract address. The second field is the contract address.

So I would almost think we would want a configuration that looks like this:

```
<gem ...>
  <gem-id> "":String </gem-id>
  <gem-addr> 0:Address </gem-addr>
  <gem-balances> .Map </gem-balances>
</gem>

syntax GemContract ::= "Gem" String
rule contract(Gem GEMID . _) => Gem GEMID
rule [[ address(Gem GEMID) => ACCTGEM ]] <gem-id> GEMID </gem-id> <gem-addr> ACCTGEM </gem-addr>
```

So that we can have tokens like `"MKR"` and `"DAI"` and `"WETH"` to identify the gems, but store the account address as a different field.

Reply:
I agree that giving  the gem a name instead of identifying it by an integer is probably better. But then, I am also pretty sure the intent of the Ilk identifier of type bytes32 is also that it's supposed to be a string, but we're representing it as an Int...

Reply:
Well, an `Address` can also be a string, so I thought we could use the named tokens as the addresses.

Reply:
Well the point would be to separate the Gem identifier and the account address, not to conflate them. Similar to how Flip is identified by it's `Ilk`, but also has a field saying which contract implements it.